docs/conditions.md

```diff
@@ -1,144 +1,100 @@
+In FluentValidation 9.0 and older, the `CascadeMode.StopOnFirstFailure` option was used to provide control over the default cascade mode at rule-level, but its use was not intuitive. There was no  `CascadeMode.Stop` option. 
+
+With `StopOnFirstFailure`,  the following would provide the example behavior described previously (stop any rule if it fails, but then continue executing at validator class-level, so that all rules are executed):
 
 ```csharp
-public PersonValidator() {
-  CascadeMode = CascadeMode.StopOnFirstFailure;
+CascadeMode = CascadeMode.StopOnFirstFailure;
 
-  RuleFor(x => x.Surname).NotNull().NotEqual("foo");
-  RuleFor(x => x.Forename).NotNull().NotEqual("foo");
-}
+RuleFor(x => x.Forename).NotNull().NotEqual("foo");
+RuleFor(x => x.MiddleNames).NotNull().NotEqual("foo");
+RuleFor(x => x.Surname).NotNull().NotEqual("foo");
 ```
-
-Using the `StopOnFirstFailure` applies only to individual rules, so by setting this on the whole validator, it is the equivalent of doing the following:
+If they all fail, you will get three validation errors. That is the equivalent of doing
 
 ```csharp
-RuleFor(x => x.Surname).Cascade(CascadeMode.StopOnFirstFailure).NotNull().NotEqual("foo");
 RuleFor(x => x.Forename).Cascade(CascadeMode.StopOnFirstFailure).NotNull().NotEqual("foo");
+RuleFor(x => x.MiddleNames).Cascade(CascadeMode.StopOnFirstFailure).NotNull().NotEqual("foo");
+RuleFor(x => x.Surname).Cascade(CascadeMode.StopOnFirstFailure).NotNull().NotEqual("foo");
 ```
+This behaviour caused a lot of confusion over the years, so the `Stop` option was introduced in FluentValidation 9.1. Using `Stop` instead of `StopOnFirstFailure`, _any_ failure at all would stop execution, so only the first failure result would be returned.
 
-That is, *both* of these rules will execute. If the `NotNull` fails on the first rule for `Surname`, the `NotEqual` will not be run. However, as the second rule for `Forename` is independent, it will also run with the same behaviour (if its `NotNull` fails, then its `NotEqual` will not run either), so by setting the validator's cascade to `StopOnFirstFailure`, you will still receive 2 validation failures (one from each rule).
-
-This behaviour has caused a lot of confusion over the years, so the `Stop` option was introduced in FluentValidation 9.1. With `Stop`, only the first failure for *any rule* will be returned. This is a true "fail-fast" behaviour.
-
-The `Stop` option was introduced rather than changing the behaviour of `StopOnFirstFailure` as this would've been a very subtle breaking change in behaviour, so we thought it was best to maintain the existing behaviour while adding a new option. `StopOnFirstFailure` has been marked as Obsolete as of FluentValidation 9.1, and will generate compiler warnings.
-
-# Dependent Rules
```

Comment:
Hi @jsabrooke The Dependent Rules section has been removed too, please can this be brought back as well